### PR TITLE
Change the like API to behave like a predicate

### LINKED ISSUE #1333: notlike operation on column
For string comparaion, `like` operation example is
```python
>>> from blaze import symbol, like, compute
>>> t = symbol('t', 'var * {name: string, city: string}')
>>> expr = like(t, name='Alice*')

>>> data = [('Alice Smith', 'New York'),
...         ('Bob Jones', 'Chicago'),
...         ('Alice Walker', 'LA')]
>>> list(compute(expr, data))
[('Alice Smith', 'New York'), ('Alice Walker', 'LA')]
```
Is there an implementation for notlike operator, similar to `sqlalchemy's` `notilike` ` ColumnOperators`?

And, inversion of `like` `~bz.like(t, name='Alice*')` doesn't seem work -- throws an error
```python
 ~bz.like(t, name='Alice*')
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-126-0c69f59e8cac> in <module>()
----> 1 ~bz.like(t, name='Alice*')

d:\apps\anaconda2\lib\site-packages\blaze\expr\core.pyc in __invert__(self)
    251
    252     def __invert__(self):
--> 253         return self._invert()
    254
    255     def __abs__(self):

d:\apps\anaconda2\lib\site-packages\blaze\expr\expressions.pyc in __getattr__(self, key)
    166             pass
    167         try:
--> 168             result = object.__getattribute__(self, key)
    169         except AttributeError:
    170             fields = dict(zip(map(valid_identifier, self.fields),
```

is this expected? Any workarounds?

### PR DESCRIPTION
**Pull request by cpcloud**
closes #1333

### CODE REVIEW

**Comment by cpcloud** [comment]
@llllllllll @cowlicks any comments here?

**Comment by cpcloud** [comment]
@llllllllll @cowlicks probably going to merge this soon

File: blaze/compute/numpy.py [hunk 1]
```diff
@@ -267,7 +267,9 @@ def compute_up(t, x, **kwargs):
 
 @dispatch(Selection, np.ndarray)
 def compute_up(sel, x, **kwargs):
-    return x[compute(sel.predicate, {sel._child: x})]
+    predicate = compute(sel.predicate, {sel._child: x})
+    cond = getattr(predicate, 'values', predicate)
```
Comment (llllllllll) [inline blaze/compute/numpy.py]: Why would this become a series?

Comment (cpcloud) [inline blaze/compute/numpy.py]: Because the `like` operation is done in the pandas backend

Comment (llllllllll) [inline blaze/compute/numpy.py]: I am trying to find where the ndarray gets coerced into a series for the `Like` to dispatch properly

Comment (cpcloud) [inline blaze/compute/numpy.py]: https://github.com/cpcloud/blaze/blob/like-bool/blaze/compute/numpy.py#L295-L301

Comment (llllllllll) [inline blaze/compute/numpy.py]: I didn't realize numpy fell back to pandas. This makes much more sense, thanks

**Comment by cpcloud** [comment]
merging on pass

**Pull request merged**